parsemsgpack.m

```diff
@@ -19,24 +20,21 @@
 
 function [obj, idx] = parse(bytes, idx)
     % masks:
-    b10000000 = 128;
-    b01111111 = 127;
-    b11000000 = 192;
-    b00111111 = 63;
-    b11100000 = 224;
-    b00011111 = 31;
-    b11110000 = 240;
-    b00001111 = 15;
+    b10000000 = uint8(128);
+    b11100000 = uint8(224);
+    b00011111 = uint8(31);
+    b11110000 = uint8(240);
+    b00001111 = uint8(15);
     % values:
-    b00000000 = 0;
-    b10010000 = 144;
-    b10100000 = 160;
+    b00000000 = uint8(0);
+    b10010000 = uint8(144);
+    b10100000 = uint8(160);
 
     currentbyte = bytes(idx);
 
     if bitand(b10000000, currentbyte) == b00000000
         % decode positive fixint
-        obj = int8(currentbyte);
+        obj = int8(currentbyte);   %if you need more performance you could try to use uint8 instead of int here
```

Comment:
I don't understand the comment. How would uint8 improve performance?

Reply:
it might bring a small performance improvement because the bytes vector is uint8 and then it doesn't need to be typecasted to int8. Since the output is positive, it could be theoretically uint8 without altering the real value.
It depends on how many positive fixints the message hat in order to have a significant performance improvement (if I remember well, in my case i could measure a small improvement, since my data was 25000 byte long).

Reply:
I see. Thank you for the analysis! If it makes a measurable performance difference, let's make it uint8.

Reply:
Done.